frontend/src/modules/LoginRedux.js

```diff
@@ -1,14 +1,81 @@
 // Action Types
 export const SET_LOGIN_STATE = "localize/SET_LOGIN_STATE";
+// URL
+export const BACKENDURL = "localhost";
+// AUTH ACTIONS
+export const AUTHENTICATED = "AUTHENTICATED";
+export const UNAUTHENTICATED = "UNAUTHENTICATED";
+export const TESTING_TOKEN_REFRESH = "TESTING_TOKEN_REFRESH";
+export const REGISTRATION_SUCCESS_MESSAGE = "REGISTRATION_SUCCESS_MESSAGE";
+//CHANGE PASSWORD ACTIONS
+export const IS_CHANGING_PASSWORD = "IS_CHANGING_PASSWORD";
+export const CHANGE_PASSWORD_SUCCESS = "CHANGE_PASSWORD_SUCCESS";
+export const CHANGE_PASSWORD_FAILURE = "CHANGE_PASSWORD_FAILURE";
 
 // Action Creators
 const setLoginState = loggedIn => ({ type: SET_LOGIN_STATE, loggedIn });
 
+let url = process.env.REACT_APP_DEV_URL || BACKENDURL;
+
+function authenticateAction(userData, dispatch, location, push) {
+  return async function() {
+    if (navigator.cookieEnabled) {
+      localStorage.setItem("ecom_token", userData.token);
+    }
+
+    if (location === "/login") {
+      push("/");
+    }
+    return dispatch({ type: AUTHENTICATED });
+  };
+}
+function registrationSuccessMessage() {
+  return { type: REGISTRATION_SUCCESS_MESSAGE };
+}
+
+function registerAction(data) {
+  return async function() {
+    let response = await fetch(`${url}/auth/users/create/`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    let responseJson = response.json();
+    return responseJson;
+  };
+}
+
+function loginAction(data) {
+  return async function() {
+    let response = await fetch(`${url}/auth/jwt/create/`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    let responseJson = await response.json();
+    return responseJson;
+  };
+}
+
+// JWT tokens are not stored in our DB
+function logoutAction() {
+  localStorage.removeItem("ecom_token");
+  return { type: UNAUTHENTICATED };
+}
+
 // Initial State
 const initialState = {
-  loggedIn: false
+  loggedIn: false,
+  authenticated: false,
+  registration_message: ""
 };
 
+const registration_message = "You have been registered successfully.";
```

Comment:
I'm not sure about this registration message at all. This came from the login project example.
I think we should simply manage these types message from the front end, not from both front end and back end. This can becomes confusing.

What do you think?

Reply:
I think that would make the most sense. Esp since connecting backend to frontend and updating it, etc to determine what language we are using just adds more that could be broken.